### PR TITLE
[CPM] Adds make targets and e2e test for migration of HA `Shoot`s

### PR DESCRIPTION
**Pull request by plkokanov**
**How to categorize this PR?**

/area control-plane-migration testing
/kind enhancement

**What this PR does / why we need it**:
This PR adds make targets for starting a second single zone HA seed cluster for the local setups (ref: [development/getting_started_locally.md](https://github.com/gardener/gardener/blob/master/docs/development/getting_started_locally.md) and [deployment/getting_started_locally.md](https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally.md)

The PR also adds an e2e test to migrate HA shoots with `spec.controlPlane.highAvailability.failureTolerance.Type: node` between single zone HA seeds.

The second commit fixes an issue with the `tear-down-kind2-env` target, which was caused because it tries to delete the `seed-local` secret which is still used by the `local` seed. Additionally, the `register-kind2-env` target was reapplying the `seed-local` secret. That secret should only be handled by the `register-local-env` target. 
A similar handling of the seed secret is also done for the `register-kind2-ha-single-zone-env` and `tear-down-kind2-ha-single-zone-env` targets.

**Which issue(s) this PR fixes**:
Part of #6529 

**Special notes for your reviewer**:

**Release note**:

```feature developer 
The local deployment of Gardener is extended so that it is now possible to create a second single zone HA `Seed`. This `Seed` can be used to test the control plane migration scenario for HA `Shoot`s. Additionally, make targets were added to trigger the control plane migration integration test with HA `Shoot`s: `test-e2e-local-migration-ha-single-zone` to test the migration locally, and `ci-e2e-kind-migration-ha-single-zone` mainly intended to be used in Gardener prow jobs.
```

```other developer
The `register-kind2-env` and `tear-down-kind2-env` will no longer try to deploy and delete the `seed-local` `Secret`. This fixes an issue where `tear-down-kind2-env` would hang as it deletes and then waits for the `seed-local` `Secret` to be deleted which can not happen as long as the `local` `Seed` which uses it still exists.  
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -336,6 +338,11 @@ kind-ha-single-zone-up: $(KIND) $(KUBECTL) $(HELM) $(YQ)
 kind-ha-single-zone-down: $(KIND)
 	./hack/kind-down.sh --cluster-name gardener-local-ha-single-zone --path-kubeconfig $(REPO_ROOT)/example/provider-local/seed-kind-ha-single-zone/base/kubeconfig
 
+kind2-ha-single-zone-up: $(KIND) $(KUBECTL) $(HELM) $(YQ)
```
Comment (ScheererJ) [inline Makefile:341]: It feels like there are now lots of make targets with very similar, marginally different actions. What do you think about unifying some of them?

Comment (plkokanov) [reply Makefile:341]: Yeah, that bothered me a bit as well, but I couldn't figure out a good way to do it, so I'm open to suggestions. One benefit of having the multiple targets is that you wouldn't have to type in any extra parameters. However, it makes it harder working on this `Makefile`. Perhaps we should also think about splitting it in separate ones?

Comment (ScheererJ) [reply Makefile:341]: I was thinking about using more variables similar to `IPFAMILY` and `KUBECONFIG` and then using those, but that is probably best handled in a separate PR. When this is merged I will try it.

**Review by ScheererJ — COMMENTED** [review]
/lgtm

**Review by rfranzke — COMMENTED** [review]
/approve

**Pull request merged**